tracer/test/Datadog.Trace.TestHelpers/CustomTestFramework.cs

```diff
@@ -275,11 +276,11 @@ public Task<T> RunAsync<T>(Func<Task<T>> action)
                 return tcs.Task;
             }
 
-            private async Task DoWork()
+            private void DoWork()
             {
                 foreach (var item in _queue.GetConsumingEnumerable())
                 {
-                    await item();
+                    item().GetAwaiter().GetResult();
```

Comment:
:stare:

Reply:
You know when there's safety instructions on the label but the expert confidently ignores them? It's one of those cases 😛

Reply:
![image](https://user-images.githubusercontent.com/18755388/234619468-180a4474-d358-48b4-9f0f-05fb4b27587b.png)